### PR TITLE
Add Led Subsystem

### PR DESCRIPTION
**Pull request by NiyanthC**

### CODE REVIEW

File: src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  init {
+    addRequirements(led, drivetrain)
+  }
+
+  override fun execute() {
+    /* led.state =
+        when (Robot.isDisabled) {
+          LedConstants.LEDMode.DISABLED
+        }
+    */
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt:22]: remove comments

File: src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    if (kotlin.math.abs(drivetrain.gyroInputs.gyroPitch.inDegrees) > 2.5) {
+      if (drivetrain.gyroInputs.gyroPitch.inDegrees > 2.5) {
+        led.state = LedConstants.LEDMode.POS_LEVELED
+      } else {
+        led.state = LedConstants.LEDMode.NEG_LEVELED
+      }
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt:29]: add a comment to switch from pitch to roll when we actually test the robot

File: src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    } else {
+      led.state =
+      when {
+        // elevator is equal to height1, height2, or height3
+        false -> LedConstants.LEDMode.OUTTAKE
+        // if holding item
+        false -> LedConstants.LEDMode.ITEM
+        Robot.isAutonomous -> LedConstants.LEDMode.AUTO
+        Robot.isDisabled -> LedConstants.LEDMode.DISABLED
+        else -> LedConstants.LEDMode.TELEOP
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt:39]: actually can we switch TELEOP -> IDLE to give further clarification

File: src/main/kotlin/com/team4099/robot2023/config/constants/LedConstants.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  enum class LEDMode {
+    IDLE,
+    ALERT,
+    ITEM,
+    POS_LEVELED,
+    NEG_LEVELED,
+    OUTTAKE,
+    AUTO,
+    TELEOP,
+    DISABLED
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2023/config/constants/LedConstants.kt:19]: remove states we don't use in LEDCommand

File: src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt (hunk 2 above)
Comment (MatthewChoulas) [reply src/main/kotlin/com/team4099/robot2023/commands/led/LedCommand.kt:29]: autobalance command currently uses pitch so i think this is fine, unless center of gravity ends up being more centered on x-axis(roll axis) and we change autobalance to use roll instead

**Pull request closed without merging**